log4j-core/src/main/java/org/apache/logging/log4j/core/config/AbstractConfiguration.java

```diff
@@ -775,12 +784,9 @@ protected void setToDefault() {
         addAppender(appender);
         final LoggerConfig rootLoggerConfig = getRootLogger();
         rootLoggerConfig.addAppender(appender, null, null);
-
-        final Level defaultLevel = Level.ERROR;
-        final String levelName =
-                contextProperties.getStringProperty(Log4jPropertyKey.CONFIG_DEFAULT_LEVEL, defaultLevel.name());
-        final Level level = Level.valueOf(levelName);
-        rootLoggerConfig.setLevel(level != null ? level : defaultLevel);
+        final String defaultLevelName = contextProperties.getStringProperty(Log4jPropertyKey.CONFIG_DEFAULT_LEVEL);
+        final Level defaultLevel = Level.toLevel(defaultLevelName, Level.ERROR);
```

Comment:
Also noticed from nullability analysis that calling `Level::valueOf` was incorrect here as it couldn't return `null`.